src/ldp/AuthenticatedLdpHandler.ts

```diff
@@ -96,9 +96,13 @@ export class AuthenticatedLdpHandler extends HttpHandler {
       err = error;
     }
 
-    const writeData = { response: input.response, description, error: err };
-
-    return this.responseWriter.handleSafe(writeData);
+    try {
+      await this.responseWriter.handleSafe({ response: input.response, description, error: err });
+    } catch (error) {
+      // Try to write the new error should the responseWriter have an issue.
+      // If this also fails the server will crash.
```

Comment:
If the responseWriter fails to write the description, isn't this a programming error then? I would suspect a hard crash in this case.

Reply:
It could also be that the ResponseWriter doesn't support writing the given description but does support writing Errors.

Reply:
For some more context: I discovered this bug by doing a request without an Accept header, causing the SimpleResourceStore to return a Quad stream, which the ResponseWriter crashed on. Now it is going to reject it in the handle function, but that is still going to throw an Error that would be uncaught without the code above.

Reply:
> It could also be that the ResponseWriter doesn't support writing the given description but does support writing Errors.

Would this be a configuration error in the server then? Would expect a generic error for this, which would then be caughted and emitted higher up.

Reply:
There are many variables here. 

Perhaps the PreferenceParser should reject input without an accept header/with an accept header that isn't supported (but then this is only relevant for certain methods so then it would need to have knowledge of that). 

Or the ResourceStore should error if there are no type preferences. But if the type is `*/*` it might output a type that is not supported by the ResponseWriter, even though it could also output a type that is supported by it (but that would definitely be a server configuration issue).

But even in the above 2 cases it could be that the ResponseWriter doesn't support the output (although in practice that should never happen since all output types should be strings or buffers).

Most of them can probably be caught but it didn't seem like a bad idea to have a catch there to make sure the server didn't fully crash just in case there was an issue. Perhaps this should even be caught in the HttpServer itself instead of here.

Reply:
My main issue is that calling very similar logic in this way after each other doesn't feel clean, and seems to suggest an underlying issue in the architecture.

> Perhaps this should even be caught in the HttpServer itself instead of here.

That's what I meant :-)

Reply:
The one disadvantage of doing it in the HttpServer is that we would duplicate our error handling code [here](https://github.com/solid/community-server/blob/master/src/ldp/http/SimpleResponseWriter.ts#L34-L39) while we already have a class that knows how to output errors.

Reply:
Can we make the HttpServer just call the response writer then?

Reply:
Certainly an option. It will make the HttpServer a bit more dependent on the internal architecture (specifically `ReponseWriter` and the `ResourceDescription` it will have to make). Not sure how problematic that is.

Reply:
Makes sense to me, since both are HTTP-dependent.
@RubenVerborgh, what do you think?

Reply:
> Can we make the HttpServer just call the response writer then?

The `ResponseWriter` is, I think, specific to LDP.

> The one disadvantage of doing it in the HttpServer is that we would duplicate our error handling code here

I think that's ok, in the sense that it will be different. Namely, if it goes all the way to the top, there's an emergency mechanism that should probably a) if a partial response has been written, aborts or b) if no response has been written, set the headers to text/plain and give a minimal description. And, in both cases, log the error.

Reply:
Changed it in the new commit

Reply:
👍